src/rdb/rdb_layout.h

```diff
@@ -80,15 +80,16 @@
 /*
  * Object ID
  *
- * The highest bit represents the object ID class. The remaining 63 bits
+ * The highest 2 bits represent the object ID class. The remaining 62 bits
  * represent the object number, which must be nonzero.
  */
 typedef uint64_t rdb_oid_t;
 
 /* Object ID class (see rdb_oid_t) */
-#define RDB_OID_CLASS_MASK	(1ULL << 63)
-#define RDB_OID_CLASS_GENERIC	(0ULL << 63)
-#define RDB_OID_CLASS_INTEGER	(1ULL << 63)
+#define RDB_OID_CLASS_MASK    (3ULL << 62)
+#define RDB_OID_CLASS_GENERIC (0ULL << 62)
+#define RDB_OID_CLASS_INTEGER (1ULL << 62)
+#define RDB_OID_CLASS_LEXICAL (2ULL << 62)
```

Comment:
[Discussion] Hoping I'm mistaken, but I think our current upgrade mechanism does not support upgrading RDB layout. Is the intention here to avoid the need to upgrade RDB layout, by assuming that existing RDBs will not use up to 2^62 objects (i.e., bit 62 of all existing object IDs is 0)? If so, we need

```
#define RDB_OID_CLASS_MASK    (3ULL << 62) // bit 63 62
#define RDB_OID_CLASS_GENERIC (0ULL << 62) //      0  0
#define RDB_OID_CLASS_INTEGER (2ULL << 62) //      1  0 <- different
#define RDB_OID_CLASS_LEXICAL (3ULL << 62) //      1  1
```

Then, no RDB upgrade procedure should be required. Is my understanding correct?